src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+                        return Pattern.compile("manual([\\\\/])[a-zA-Z]+$").matcher(file.getPath())
+                                .find();
+                    }
+
+                    @Override
+                    public boolean accept(File dir, String name) {
+                        return true;
+                    }
+                }
+        );
+        caseDirs.remove(0);
+        return caseDirs;
+    }
+
+    private void compileTestClasses() throws Exception {
+
+        PageObjectsGenerator pog = initPog(
+                jsonSourcePath,
+                url,
+                checkLocatorUniqueness,
+                forceGenerateFiles);
+
+        pog.generatePageObjects();
+
+        JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+        StandardJavaFileManager standardJavaFileManager = compiler.getStandardFileManager(null, null, null);
+        String[] options = new String[]{"-d", "target/test-classes"};
+        File[] files = new File[]{
+                new File(inputJavaFileTest),
+                new File(inputJavaFileManual)
+        };
+
+        JavaCompiler.CompilationTask compilationTask = compiler.getTask(null, null, null,
+                Arrays.asList(options),
+                null,
+                standardJavaFileManager.getJavaFileObjects(files));
+        boolean compileSuccess = compilationTask.call();
+        assertEquals(true, compileSuccess);
+
+        CustomClassLoader classLoader = new CustomClassLoader();
+        testClass = classLoader.loadClass(this.generatedClassFullName);
+        manualClass = classLoader.loadClass(this.manualClassFullName);
+
+    }
+
+    @Test
+    public void runForPackage() throws Exception {
+        classesList.sort((pathToJavaFile1, pathToJavaFile2) -> pathToJavaFile1.getSmallPath()
+                .endsWith("page/") ? -1 : pathToJavaFile2.getSmallPath().endsWith("page/") ? 1 : 0);
+
+        for (TestClassDTO classDTO : classesList) {
+            this.inputJavaFileTest = RESOURCE_DIR + TEST_DIR + classDTO.getSmallPath()
+                    .split("/")[1] + "/" + classDTO.getSmallName() + ".java";
+            this.inputJavaFileManual = RESOURCE_DIR + MANUAL_DIR + classDTO.getSmallPath()
+                    + classDTO.getSmallName() + ".java";
+            this.generatedClassFullName = TEST_CLASS_PACKAGE_NAME + classDTO.getSmallPath()
+                    .split("/")[1] + "." + classDTO.getSmallName();
+            this.manualClassFullName = MANUAL_CLASS_PACKAGE_NAME + classDTO.getSmallPath().
+                    replace("/", ".") + classDTO.getSmallName();
+            this.jsonSourcePath = classDTO.getJsonPath();
+            this.url = classDTO.getTestURL();
+            this.checkLocatorUniqueness = classDTO.isCheckLocatorUniqueness();
+            this.forceGenerateFiles = classDTO.isForceGenerateFiles();
+            runTestCase();
+        }
+    }
+
+    private void runTestCase() throws Exception {
+        compileTestClasses();
```

Comment:
Please, modify all methods to accept parameters and return values. Do not be afraid to create new model classes to store results of methods execution. To make code readable, avoid usage of testclass fields without reason. Current procedural implementation makes code unreadable, even you will not be able to support this class without strong mind effort.